lib/rubocop/cop/lint/redundant_type_conversion.rb

```diff
@@ -69,10 +70,12 @@ module Lint
       #   foo.to_s
       #
       #   # bad - chaining a conversion to a method that is expected to return the same type
-      #   inspect.to_s
+      #   foo.inspect.to_s
+      #   foo.to_json.to_s
```

Comment:
I've also seen `"#{foo.to_json}"` used [out in the wild](https://github.com/search?q=%2F%22%23%5C%7B%5Cw%2B%5C.to_json%5C%7D%22%2F%20language%3Aruby&type=code), I might open a PR to register this offense in Style/RedundantInterpolation.